Building avif support fails on Ubuntu 18.04/gcc 7.4.0

<!-- IMPORTANT
Bug reports that do not make an effort to help the developers will be closed without notice.
Make sure that this bug has not already been opened and/or closed by searching the issues on GitHub, as duplicate bug reports will be closed.
A bug report simply stating that Darktable crashes is unhelpful, so please fill in most of the items below and provide detailed information.
-->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
Initial attempts to build with avif support failed with the compiler error below.  Adding  a         -Wno-uninitialized compiler flag enabled the building to complete and the executable seemed to work ok at both importing and exporting .avif files.  However the aom encoder was very slow and went into an infinite loop at high resolutions (from reading around this seems to be a recognised aom issue).  rav1e was fine as an encoder.

/home/jim/darktable/src/common/imageio_avif.c: In function ‘dt_imageio_open_avif
’:
/home/jim/darktable/src/common/imageio_avif.c:257:3: error: ‘decoder’ may be use
d uninitialized in this function [-Werror=maybe-uninitialized]
   avifDecoderDestroy(decoder);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jim/darktable/src/common/imageio_avif.c: In function ‘dt_imageio_avif_read
_color_profile’:
/home/jim/darktable/src/common/imageio_avif.c:521:3: error: ‘decoder’ may be use
d uninitialized in this function [-Werror=maybe-uninitialized]
   avifDecoderDestroy(decoder);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 41%] Building C object src/CMakeFiles/lib_darktable.dir/gui/camera_import_dial
og.c.o
cc1: all warnings being treated as errors
src/CMakeFiles/lib_darktable.dir/build.make:2812: recipe for target 'src/CMakeFi
les/lib_darktable.dir/common/imageio_avif.c.o' failed


**To Reproduce**
<!-- Provide detailed steps that can reproduce the behavior, such as:
1. Go to '...'
2. Click on '...'
3. Scroll down to '...'
4. See error
-->


**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
to build without error warnings

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->


**Platform (please complete the following information):**
 - Darktable Version: <!-- [e.g. 2.6.0] -->git latest
 - OS: <!-- [e.g. Windows 8.1, Gentoo Linux] -->Ubuntu 16.04 gcc 7.4
 - <!-- OpenCL activated or no? -->not relevant
 - <!-- Which graphics card and driver version -->not relevant


**Additional context**
<!-- Add any other context about the problem here, for example:
 - Can you reproduce with another Darktable version?
 - Can you reproduce with a RAW or Jpeg or both?
 - Are the steps above reproduce with a fresh edit (removing history)?
 - Attach an XMP if this is necessary
 - Did you compile Darktable yourself? If so which compiler was used, with what options?
 - Is the issue still present using an empty/new config-dir
-->

--- a/src/common/imageio_avif.c
+++ b/src/common/imageio_avif.c
@@ -105,6 +105,7 @@ dt_imageio_retval_t dt_imageio_open_avif(dt_image_t *img,
   dt_imageio_retval_t ret;
   avifROData raw = AVIF_DATA_EMPTY;
   avifImage *avif = NULL;
+  avifDecoder *decoder = NULL;
   avifResult result;
 
   ret = read_image(filename, &raw);
@@ -124,7 +125,7 @@ dt_imageio_retval_t dt_imageio_open_avif(dt_image_t *img,
     goto out;
   }
 
-  avifDecoder *decoder = avifDecoderCreate();
+  decoder = avifDecoderCreate();
   if (decoder == NULL) {
     dt_print(DT_DEBUG_IMAGEIO,
              "Failed to create AVIF decoder for image [%s]\n",
@@ -264,6 +265,7 @@ dt_imageio_retval_t dt_imageio_avif_read_color_profile(const char *filename, str
 {
   dt_imageio_retval_t ret;
   avifROData raw = AVIF_DATA_EMPTY;
+  avifDecoder *decoder = NULL;
   avifResult result;
 
   ret = read_image(filename, &raw);
@@ -283,7 +285,7 @@ dt_imageio_retval_t dt_imageio_avif_read_color_profile(const char *filename, str
     goto out;
   }
 
-  avifDecoder *decoder = avifDecoderCreate();
+  decoder = avifDecoderCreate();
   if (decoder == NULL) {
     dt_print(DT_DEBUG_IMAGEIO,
              "Failed to create AVIF decoder for image [%s]\n",

